config/retry/backoff.go

```diff
@@ -413,3 +414,34 @@ func (b *Backoffer) CheckKilled() error {
 	}
 	return nil
 }
+
+// IsFakeRegionError returns true if err is fake region error.
+func IsFakeRegionError(err *errorpb.Error) bool {
+	return err != nil && err.GetEpochNotMatch() != nil && len(err.GetEpochNotMatch().CurrentRegions) == 0
+}
+
+// MayBackoffOrFailFastForRegionError handles the region error for some cases:
+// - If the `regionError` should not retry, returns an error.
+// - If the `regionError` can retry immediately, it returns nil.
+// - If the `regionError` can retry but need to do backoff, it backoff first and returns nil.
+// - If the backoff in the above case fails (.i.e reaches the max backoff limit), it returns the error.
+func MayBackoffOrFailFastForRegionError(regionErr *errorpb.Error, bo *Backoffer) error {
+	if regionErr == nil {
+		return nil
+	}
+
+	if regionErr.GetUndeterminedResult() != nil {
```

Comment:
The undetermined error should be returned in following cases:
1. 2pc protocol is used, and the undetermined error is returned in the response of primary key commit request.
2. Async commit protocol is used, and  the undetermined error is returned in either responess of prewrite requests.

For other cases, the undetermined error could be handled in the same way as rpc timeout error, so we may not need to adapt `MayBackoffOrFailFastForRegionError` in other commands like `TxnHeartBeat`, `PessimsiticLock`.